routers/repo/issue_timetrack.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+	"net/http"
+	"strconv"
+)
+
+// AddTimeManual tracks time manually
+func AddTimeManual(c *context.Context) {
+	hours, err := strconv.ParseInt(c.Req.PostForm.Get("hours"), 10, 64)
+	if err != nil {
+		if c.Req.PostForm.Get("hours") != "" {
+			c.Handle(http.StatusInternalServerError, "hours is not numeric", err)
```

Comment:
It would also be great to refactor in a separate function. I have made an example witih time.Duration or with strconv: https://play.golang.org/p/5W1Hsfqoc-